packages-content-model/roosterjs-content-model-editor/lib/publicApi/table/applyTableBorderFormat.ts

```diff
@@ -0,0 +1,149 @@
+import { extractBorderValues } from '../../domUtils/borderValues';
+import { formatWithContentModel } from '../utils/formatWithContentModel';
+import { getFirstSelectedTable } from '../../modelApi/selection/collectSelections';
+import { getSelectedCells } from '../../modelApi/table/getSelectedCells';
+import { parseValueWithUnit } from 'roosterjs-content-model-dom';
+import { updateTableCellMetadata } from '../../domUtils/metadata/updateTableCellMetadata';
+import type { BorderOperations } from '../../publicTypes/enum/BorderOperations';
+import type { IContentModelEditor } from '../../publicTypes/IContentModelEditor';
+import type { Border } from '../../publicTypes/interface/Border';
+import type { ContentModelTableCell } from 'roosterjs-content-model-types';
+
+/**
+ * UNFINISHED - only All Borders is implemented
+ * @param editor The editor instance
+ * @param border The border to apply
+ * @param operation The operation to apply
+ */
+export default function applyTableBorderFormat(
+    editor: IContentModelEditor,
+    border: Border,
+    operation: BorderOperations
+) {
+    formatWithContentModel(editor, 'tableBorder', model => {
+        const [tableModel] = getFirstSelectedTable(model);
+
+        if (tableModel) {
+            const sel = getSelectedCells(tableModel);
+            const perimeter = {
+                Top: false,
+                Bottom: false,
+                Left: false,
+                Right: false,
+            };
+
+            // Create border format with table format as backup
+            let borderFormat = '';
+            const format = tableModel.format;
+            const { width, style, color } = border;
+            const extractedBorder = extractBorderValues(format.borderTop);
```

Comment:
I just realized that this format is from table, and we only check its top border now. What if table cell has different border than table, and what if the borders in a cell are different, e.g. top vs bottom?

Reply:
Can we just force the type `Border` must have all properties have value? Then we just need to read the value from the parameter, and no need to read from current table. Caller code can decide what value to pass in.

For example, if user has specified border color, but didn't specify style and width, caller code pass in the specified color together with default width and style? Then in this function we can always assume parameter already passes in enough information.

Reply:
It is just a fallback value. Forcing changes in Border affects Image Borders too.

Reply:
I could make the Border type Required<Border> for the function.